test/gtest/tl/mlx5/test_tl_mlx5_qps.h

```diff
@@ -0,0 +1,175 @@
+
+typedef ucc_status_t (*ucc_tl_mlx5_create_rc_qp_fn_t)(
+    struct ibv_context *ctx, struct ibv_pd *pd, struct ibv_cq *cq, int ib_port,
+    int tx_depth, ucc_tl_mlx5_qp_t *qp, uint32_t *qpn, ucc_base_lib_t *lib);
+
+class test_tl_mlx5_rc_qp : public test_tl_mlx5_qp {
+  public:
+    ucc_tl_mlx5_qp_t               qp;
+    uint32_t                       qpn;
+    ucc_tl_mlx5_qp_t               umr_qp;
+    ucc_tl_mlx5_ib_qp_conf_t       umr_qp_conf;
+    int                            tx_depth;
+    ucc_tl_mlx5_create_rc_qp_fn_t  create_rc_qp;
+    ucc_tl_mlx5_qp_connect_fn_t    qp_connect;
+    ucc_tl_mlx5_create_umr_qp_fn_t create_rc_umr_qp;
+
+    virtual void SetUp()
+    {
+        test_tl_mlx5_qp::SetUp();
+
+        create_rc_qp = (ucc_tl_mlx5_create_rc_qp_fn_t)dlsym(
+            tl_mlx5_so_handle, "ucc_tl_mlx5_create_rc_qp");
+        ASSERT_EQ(nullptr, dlerror());
+
+        qp_connect = (ucc_tl_mlx5_qp_connect_fn_t)dlsym(
+            tl_mlx5_so_handle, "ucc_tl_mlx5_qp_connect");
+        ASSERT_EQ(nullptr, dlerror());
+
+        create_rc_umr_qp = (ucc_tl_mlx5_create_umr_qp_fn_t)dlsym(
+            tl_mlx5_so_handle, "ucc_tl_mlx5_create_umr_qp");
+        ASSERT_EQ(nullptr, dlerror());
+
+        qp.qp       = NULL;
+        umr_qp.qp   = NULL;
+        tx_depth    = 4;
+        umr_qp_conf = qp_conf;
+    }
+
+    ~test_tl_mlx5_rc_qp()
+    {
+        if (qp.qp) {
+            ibv_destroy_qp(qp.qp);
+        }
+        if (umr_qp.qp) {
+            ibv_destroy_qp(umr_qp.qp);
+        }
+    }
+
+    void create_qp()
+    {
+        GTEST_ASSERT_EQ(
+            create_rc_qp(ctx, pd, cq, port, tx_depth, &qp, &qpn, &lib), UCC_OK);
+    }
+
+    void create_umr_qp()
+    {
+        GTEST_ASSERT_EQ(
+            create_rc_umr_qp(ctx, pd, cq, port, &umr_qp.qp, &umr_qp_conf, &lib),
+            UCC_OK);
+    }
+
+    void connect_qp_loopback()
+    {
+        GTEST_ASSERT_EQ(
+            qp_connect(qp.qp, qpn, port_attr.lid, port, &qp_conf, &lib),
+            UCC_OK);
+    };
+};
+
+typedef ucc_status_t (*ucc_tl_mlx5_init_dct_fn_t)(
+    struct ibv_pd *pd, struct ibv_context *ctx, struct ibv_cq *cq,
+    struct ibv_srq *srq, uint8_t port_num, struct ibv_qp **dct_qp,
+    uint32_t *qpn, ucc_tl_mlx5_ib_qp_conf_t *qp_conf, ucc_base_lib_t *lib);
+
+typedef ucc_status_t (*ucc_tl_mlx5_init_dci_fn_t)(
+    ucc_tl_mlx5_dci_t *dci, struct ibv_pd *pd, struct ibv_context *ctx,
+    struct ibv_cq *cq, uint8_t port_num, int tx_depth,
+    ucc_tl_mlx5_ib_qp_conf_t *qp_conf, ucc_base_lib_t *lib);
+
+typedef ucc_status_t (*ucc_tl_mlx5_create_ah_fn_t)(struct ibv_pd * pd,
+                                                   uint16_t        lid,
+                                                   uint8_t         port_num,
+                                                   struct ibv_ah **ah_ptr,
+                                                   ucc_base_lib_t *lib);
+
+class test_tl_mlx5_dc : public test_tl_mlx5_qp {
```

Comment:
is "test_tl_mlx5_dc" class used some where? i dont see it

Reply:
This class is used for the tests in `test_tl_mlx5_qps.cc`
For now the tests on dc are only about qp creation. We will add more tests later when we will implement support for dc qps (for now, only rc qps are fully supported)